homeassistant/components/switcher_kis/__init__.py

```diff
@@ -1,85 +1,181 @@
+            device.device_type.value,
+            device.device_type.hex_rep,
+        )
 
-    await v2bridge.start()
+        wrapper = hass.data[DOMAIN][DATA_DEVICE][
+            device.device_id
+        ] = SwitcherDeviceWrapper(hass, entry, device)
+        hass.async_create_task(wrapper.async_setup())
 
-    async def async_stop_bridge(event: EventType) -> None:
-        """On Home Assistant stop, gracefully stop the bridge if running."""
-        await v2bridge.stop()
+    async def platforms_setup_task() -> None:
+        # Must be ready before dispatcher is called
+        for platform in PLATFORMS:
+            await hass.config_entries.async_forward_entry_setup(entry, platform)
 
-    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, async_stop_bridge)
+        discovery_task = hass.data[DOMAIN].pop(DATA_DISCOVERY, None)
+        if discovery_task is not None:
+            discovered_devices = await discovery_task
+            for device in discovered_devices.values():
+                on_device_data_callback(device)
 
-    try:
-        device_data = await wait_for(v2bridge.queue.get(), timeout=10.0)
-    except (Asyncio_TimeoutError, RuntimeError):
-        _LOGGER.exception("Failed to get response from device")
-        await v2bridge.stop()
-        return False
-    hass.data[DOMAIN] = {DATA_DEVICE: device_data}
+        await async_start_bridge(hass, on_device_data_callback)
 
-    hass.async_create_task(async_load_platform(hass, "switch", DOMAIN, {}, config))
-    hass.async_create_task(async_load_platform(hass, "sensor", DOMAIN, {}, config))
+    hass.async_create_task(platforms_setup_task())
 
     @callback
-    def device_updates(timestamp: datetime | None) -> None:
-        """Use for updating the device data from the queue."""
-        if v2bridge.running:
-            try:
-                device_new_data = v2bridge.queue.get_nowait()
-                if device_new_data:
-                    async_dispatcher_send(
-                        hass, SIGNAL_SWITCHER_DEVICE_UPDATE, device_new_data
-                    )
-            except QueueEmpty:
-                pass
-
-    async_track_time_interval(hass, device_updates, timedelta(seconds=4))
+    async def stop_bridge(event: Event) -> None:
+        await async_stop_bridge(hass)
+
+    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, stop_bridge)
 
     return True
+
+
+class SwitcherDeviceWrapper(update_coordinator.DataUpdateCoordinator):
+    """Wrapper for a Switcher device with Home Assistant specific functions."""
+
+    def __init__(
+        self, hass: HomeAssistant, entry: ConfigEntry, device: SwitcherBase
+    ) -> None:
+        """Initialize the Switcher device wrapper."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=device.name,
+            update_interval=timedelta(seconds=MAX_UPDATE_INTERVAL_SEC),
+        )
+        self.hass = hass
+        self.entry = entry
+        self.data: SwitcherBase = device
```

Comment:
This typing can be infered
```suggestion
        self.data = device
```